### PR TITLE
Set up calling acronym data from store. Add unit tests

### PR DESCRIPTION
**Pull request by loneil**
# Description
https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-765

Add in the vuex store module for users and return the acronym data (still dummy in this iterative PR) from there rather than in the component.
Add required unit tests for stuff added as part of this ticket.

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-k8vopl-tools.pathfinder.gov.bc.ca/dashboard?id=nr-get-token-pr-134

File: app/frontend/src/components/ApplicationList.vue [hunk 1]
```diff
@@ -46,18 +46,25 @@
 </template>
 
 <script>
+import { mapGetters } from 'vuex';
+import { mapActions } from 'vuex';
```
Comment (jujaga) [inline app/frontend/src/components/ApplicationList.vue]: `import { mapActions, mapGetters } from 'vuex';`

File: app/frontend/src/store/modules/user.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        state.acronyms = acronyms;
+      }
+    }
+  },
+  actions: {
+    /**
+    * @function getUserAcronyms
+    * Fetch the acronyms the current user has access to from the DB
+    * @param {object} context The store context
+    */
```
Comment (jujaga) [inline app/frontend/src/store/modules/user.js]: Missing a space on all the JSDocs after the /**

File: app/frontend/src/services/userService.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+export default {
+  /** WORK IN PROGRESS
+  * @function getUserAcronyms
+  * Fetch the acronyms the current user has access to from the DB
+  * @param {string} userGuid The current user
+  */
```
Comment (jujaga) [inline app/frontend/src/services/userService.js]: Missing a space on all the JSDocs after the /**

File: app/frontend/tests/unit/services/userService.spec.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import userService from '@/services/userService';
+
+describe('getUserAcronyms()', () => {
+
+  it('calls `getUserAcronyms() endpoint', async () => {
```
Comment (jujaga) [inline app/frontend/tests/unit/services/userService.spec.js]: Let's describe the function without ` and ()

File: app/frontend/tests/unit/store/modules/user.actions.spec.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  let store;
+  beforeEach(() => {
+    const localVue = createLocalVue();
+    localVue.use(Vuex);
+
+    store = new Vuex.Store(cloneDeep(userStore));
+  });
+
+  const spy = jest.spyOn(userService, 'getUserAcronyms');
+  jest.mock('../../../../src/services/userService');
```
Comment (jujaga) [inline app/frontend/tests/unit/store/modules/user.actions.spec.js]: This line might not be needed as you're already spying on getUserAcronyms (which by definition forces that function to be mock intercepted)

File: app/frontend/tests/unit/store/modules/user.actions.spec.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+  afterEach(() => {
```
Comment (jujaga) [inline app/frontend/tests/unit/store/modules/user.actions.spec.js]: Let's use beforeEach more if possible to cleanse test state

File: app/frontend/tests/unit/store/modules/user.actions.spec.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    await store.dispatch('getUserAcronyms');
+
+    expect(store.getters.acronyms).toEqual(mockAcronyms);
+  });
+
+  it('handles a blank return', async () => {
+    userService.getUserAcronyms.mockResolvedValue(undefined);
+    await store.dispatch('getUserAcronyms');
+
+    expect(store.getters.acronyms).toEqual([]);
```
Comment (jujaga) [inline app/frontend/tests/unit/store/modules/user.actions.spec.js]: Side note: An alternative check can be `expect(store.getters.acronyms).toEqual(expect.any(Array));` if you just want to make sure it's an array.

File: app/frontend/tests/unit/store/modules/user.spec.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import { createLocalVue } from '@vue/test-utils';
+import Vuex from 'vuex';
+import userStore from '@/store/modules/user';
+import { cloneDeep } from 'lodash';
```
Comment (jujaga) [inline app/frontend/tests/unit/store/modules/user.spec.js]: This should be on line 1

File: app/frontend/src/components/ApplicationList.vue (hunk 1 above)
Comment (loneil) [reply app/frontend/src/components/ApplicationList.vue]: Fixed

File: app/frontend/tests/unit/store/modules/user.actions.spec.js (hunk 5 above)
Comment (loneil) [reply app/frontend/tests/unit/store/modules/user.actions.spec.js]: Ah didn't know, thanks

File: app/frontend/tests/unit/store/modules/user.actions.spec.js (hunk 6 above)
Comment (loneil) [reply app/frontend/tests/unit/store/modules/user.actions.spec.js]: k

File: app/frontend/tests/unit/store/modules/user.actions.spec.js (hunk 7 above)
Comment (loneil) [reply app/frontend/tests/unit/store/modules/user.actions.spec.js]: Going to check equality of an equal array here but good to know

File: app/frontend/tests/unit/store/modules/user.spec.js (hunk 8 above)
Comment (loneil) [reply app/frontend/tests/unit/store/modules/user.spec.js]: Fixed

**Pull request merged**